### PR TITLE
Feat/distinctField prop

### PR DESCRIPTION
**Pull request by anik-ghosh-au7**
Support and examples added for `distinctField `& `distinctFieldConfig` props in the following components - which returns distinct results based on a specified field.

- React-SearchBox
- Vue-SearchBox
- React-Native-SearchBox

Notion Link - https://www.notion.so/appbase/distinctField-prop-f14649fd05c544a993cbee11e1846c8c

Docs PR link - https://github.com/appbaseio/Docs/pull/186

Demo - 
https://www.loom.com/share/cd033076e0574ebf8b739fc14d7a2b3c

### CODE REVIEW

File: packages/native/examples/with-distinct-filed-prop/App.js [hunk 1]
```diff
@@ -0,0 +1,214 @@
+import React, { useState, useRef } from 'react';
```
Comment (bietkul) [inline packages/native/examples/with-distinct-filed-prop/App.js:1]: @anik-ghosh-au7 You've created a new example with `distinctField` prop but how is it different from the `basic` demo visually? There should be some meaningful use of that data that you receive after using the new props. You can use it in `SearchComponent` and group the results by authors and display the `inner_hits` in UI. Get approval before making this change everywhere.

File: packages/react-searchbox/examples/with-distinct-field-prop/src/App.js [hunk 2]
```diff
@@ -0,0 +1,227 @@
+import React from 'react';
```
Comment (bietkul) [inline packages/react-searchbox/examples/with-distinct-field-prop/src/App.js:1]: @anik-ghosh-au7 Example is not meaningful. Remove the author filter and follow the `native` example.

File: packages/searchbase/src/utils.js [hunk 3]
```diff
@@ -446,6 +446,8 @@ export const searchBaseMappings = {
   suggestions: 'suggestions',
   queryId: 'queryId',
   recentSearches: 'recentSearches',
+  distinctField: 'distinctField',
+	distinctFieldConfig: 'distinctFieldConfig',
```
Comment (bietkul) [inline packages/searchbase/src/utils.js]: @anik-ghosh-au7 Why formatting issues?

File: packages/vue-searchbox/examples/with-distinct-field-prop/src/App.vue [hunk 4]
```diff
@@ -0,0 +1,217 @@
+<template>
```
Comment (bietkul) [inline packages/vue-searchbox/examples/with-distinct-field-prop/src/App.vue:1]: @anik-ghosh-au7 Follow react example.

File: packages/vue-searchbox/src/components/SearchBox.jsx [hunk 5]
```diff
@@ -115,6 +117,11 @@ const SearchBox = {
 		return this.state;
 	},
 	mounted() {
+		if (this.aggregationField !== undefined) {
```
Comment (bietkul) [inline packages/vue-searchbox/src/components/SearchBox.jsx]:
```suggestion
		if (this.aggregationField) {
```

**Review by bietkul — CHANGES_REQUESTED** [review]
@anik-ghosh-au7 
1. In the RN example, why are getting the following errors/warnings?
alt img
alt img2

2. Typescript definition is missing in `searchbase` library

**Comment by anik-ghosh-au7** [comment]
Updated examples Demo for distinctField props in - 
 - React SearchBox
 - Vue SearchBox
 - React-native SearchBox

https://www.loom.com/share/ae2673c11c3a4f36819ba04d6159db6f

File: packages/native/examples/with-distinct-filed-prop/App.js [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+} from 'react-native';
+
+const renderResultItem = ({ item }) => {
+  const otherBooks = item.inner_hits
+    ? item.inner_hits.other_books.hits.hits
+    : [];
+  const updatedBooks = otherBooks.filter(
+    book => book._source.original_title !== item.original_title
+  );
+  // console.log('updatedBooks ==>> ', updatedBooks);
```
Comment (bietkul) [inline packages/native/examples/with-distinct-filed-prop/App.js]: @anik-ghosh-au7 Remove console

File: packages/native/examples/with-distinct-filed-prop/App.js [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+                key={item._id + `_${index}`}
+                name="star"
+                size={24}
+                color="gold"
+              />
+            ))}
+          <Text style={styles.rating}>({item.average_rating} avg)</Text>
+        </View>
+        <Text>Pub {item.original_publication_year}</Text>
+        {updatedBooks.length > 0 && (
```
Comment (bietkul) [inline packages/native/examples/with-distinct-filed-prop/App.js]:
```suggestion
        {updatedBooks.length && (
```

File: packages/native/examples/with-distinct-filed-prop/App.js [hunk 8]
```diff
@@ -0,0 +1,275 @@
...
+                  <View style={styles.otherBooksContainer}>
+                    <Image
+                      style={styles.otherBooksImage}
+                      source={{
+                        uri: book._source.image
+                      }}
+                      resizeMode="contain"
+                    />
+                    <Text style={styles.otherBooksText}>
+                      {book._source.original_title.length > 12
```
Comment (bietkul) [inline packages/native/examples/with-distinct-filed-prop/App.js]: @anik-ghosh-au7 You can avoid this check. `substring` method is safe to be used

File: packages/react-searchbox/examples/with-distinct-field-prop/src/App.js [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+                                    </span>
+                                  </div>
+                                </div>
+                                <span className="pub-year">
+                                  Pub {item.original_publication_year}
+                                </span>
+                              </div>
+                            </div>
+                          </div>
+                          {updatedBooks.length > 0 && (
```
Comment (bietkul) [inline packages/react-searchbox/examples/with-distinct-field-prop/src/App.js]:
```suggestion
                          {updatedBooks.length && (
```

File: packages/react-searchbox/examples/with-distinct-field-prop/src/App.js [hunk 10]
```diff
@@ -0,0 +1,248 @@
...
+                                                .original_title
+                                            }
+                                          </h6>
+                                        </div>
+                                      </div>
+                                    )}
+                                  </div>
+                                </div>
+                                <div className="flex col">
+                                  <div className="flex row">
```
Comment (bietkul) [inline packages/react-searchbox/examples/with-distinct-field-prop/src/App.js]: Use map and avoid iterating the same functionality.

**Review by bietkul — CHANGES_REQUESTED** [review]

File: packages/native/examples/with-distinct-filed-prop/App.js (hunk 8 above)
Comment (anik-ghosh-au7) [reply packages/native/examples/with-distinct-filed-prop/App.js]: used `numberOfLines` prop

**Review by bietkul — APPROVED** [review]

**Pull request merged**